packages/app/src/systems/CRX/background/services/DatabaseEvents.ts

```diff
@@ -51,5 +52,35 @@ export class DatabaseEvents {
         ])
       );
     });
+
+    this.databaseObservable.on('accounts:update', async (updateEvent) => {
+      // Broadcast only if it's the current account
+      if (!updateEvent.obj.isCurrent) return;
+
+      const [currentAccount, connections] = await Promise.all([
+        AccountService.getCurrentAccount(),
```

Comment:
You can get the data directly from the update event.obj removing the need to query the data again.

Reply:
_that makes sense_